### DISCUSSION TITLE
Add feature in ps_gdpr module for PS 1.7

### DISCUSSION THREAD

**Discussion by goncalosribeiro** [PrestaShop/PrestaShop#34676] [Ideas]
**Is your feature request related to a problem? Please describe.**
The Official GDPR Compliance by PrestaShop module for PS 1.7. allows for a registered client to download the saved personal information (PDF or CSV), in their personal information page. How about for not registered clients?

**Describe the solution you'd like**
In the module configuration page, in the second item "Manage personal data" (or some similar translation), by searching for a client, and clicking on it, it is possible to see below all the saved data of that client (registered or unregistered). It should be also possible to download that information, in PDF or CSV, of that specific client, in the Backoffice module configuration page. There can be clients asking fot that information.

---
I suggest is to have a data download button on the backoffice, in the module configuration page, where the guest client information is presented, in case that client wants the data. In these cases the client must ask for the information via email or contact form.

The only buttons available, when searching for a client, is to delete data or download invoices, on each client found in the search. Click on one client and below this it is presented the same information that is downloaded when a registered client downloads on its personal information page. 

The button I suggest should appear somewhere over this information or on each client window.

**Comment by khouloudbelguith** [comment]
Hi @gonalo123,

What do you mean by unregistered customer?
Is it a Guest customer?

**Comment by goncalosribeiro** [comment]
Yes. Precisely.

**Comment by khouloudbelguith** [comment]
@gonalo123, so your suggestion is to add the possibility for "Guest" customer to download their saved personal information from the Front-end.

**Comment by goncalosribeiro** [comment]
No.
What I suggest is to have a data download button on the backoffice, in the module configuration page, where the guest client information is presented, in case that client wants the data. In these cases the client must ask for the information via email or contact form.
The only buttons available, when searching for a client, is to delete data or download invoices, on each client found in the search. Click on one client and below this it is presented the same information that is downloaded when a registered client downloads on its personal information page. The button I suggest should appear somewhere over this information or on each client window.

**Comment by khouloudbelguith** [comment]
Hi @gonalo123, thanks for this clarifications.
@colinegin what do you think of his suggestion?

**Comment by colinegin** [comment]
Hello,

Thanks for your suggestion @gonalo123 ! It's a great idea, but I need to discuss it with the module developer. I've just changed the title of this issue as it is related to the GDPR module and not legal compliance module.

I will keep you updated ! In the meantime, if you are able to code this feature, feel free to do a PR !

**Comment by goncalosribeiro** [comment]
Yes, that is correct, not the ps_legalcompliance. I suggest you remove the FO, tag also because this is a backoffice feature. 

As for coding, I don´t think I could do it. My capabilities wouldn't allow me to do it in less than a year. :)

**Comment by hibatallahAouadni** [comment]
Ping @PrestaShop/product-team friendly reminder :wink:
